automation/jleechanorg_pr_automation/automation_utils.py

```
@@ -247,21 +242,57 @@ def execute_subprocess_with_timeout(cls, command: list, timeout: int = None,
+                data = json.load(f)
+                fcntl.flock(f.fileno(), fcntl.LOCK_UN)  # Unlock
+                return data
+        except (FileNotFoundError, json.JSONDecodeError):
+            return {}
 
     @classmethod
-    def safe_write_json(cls, file_path: Path, data: dict) -> None:
-        """Atomically write JSON file with file locking (delegates to json_manager)"""
-        if not json_manager.write_json(str(file_path), data):
-            raise RuntimeError(f"Failed to write JSON file {file_path}")
+    def safe_write_json(cls, file_path: Path, data: dict):
```

Ruff (0.14.11)

Missing return type annotation for classmethod `safe_write_json`

Add return type annotation: `None`

(ANN206)